utils/notificationhubServicePartition.ts

```diff
@@ -44,22 +45,19 @@ export const testShaForPartitionRegex = (
  */
 export const getNotificationHubPartitionConfig = (envConfig: IConfig) => (
   sha: InstallationId
-): NotificationHubConfig => {
-  const partition = envConfig.AZURE_NOTIFICATION_HUB_PARTITIONS.find(p =>
-    testShaForPartitionRegex(p.partitionRegex, sha)
-  );
-
-  if (partition) {
-    const connectionString = partition.endpoint;
-
-    return {
-      AZURE_NH_ENDPOINT: connectionString,
+): NotificationHubConfig =>
```

Comment:
Just a minor style comment: instead of throwing an exception, what if we return `Option<NotificationHubConfig>`? 
Benefit:
* the signature communicates that the search may fail
* we delegate the caller to decide how to handle such failure

Reply:
Since we are pretty confident that this will never happen (because of the checks on the partition array) I thought it was not necessary to propagate an `Option<>`.
What's the benefit of having it in this particular case?

Reply:
> this will never happen

Don't let me quote Murphy's law 😝. 

Benefits are on being explicit about what the result could be, and let the language drive you into writing safer code. 

`throw`s are hidden from the compiler, so to handle them correctly is up to developers' discipline (which is an error-prone thing).
`Option`s (or `Either`s) don't let you read the value unless you handle all possible cases.